llm/lorax/README.md

```diff
@@ -0,0 +1,113 @@
+  docker run --gpus all --shm-size 1g -p 8080:80 -v ~/data:/data \
+    ghcr.io/predibase/lorax:latest \
+    --model-id $MODEL_ID
+```
+
+In the above example, we're asking SkyPilot to provision an AWS instance with 1 Nvidia A10G GPU and at least 32GB of RAM. Once the node is provisioned,
+SkyPilot will launch the LoRAX server using our latest pre-built [Docker image](https://github.com/predibase/lorax/pkgs/container/lorax).
+
+Let's launch our LoRAX job:
+
+```shell
+sky launch -c lorax-cluster lorax.yaml
+```
+
+By default, this config will deploy `Mistral-7B-Instruct`, but this can be overridden by running `sky launch` with the argument `--env MODEL_ID=<my_model>`.
+
+**NOTE:** This config will launch the instance on a public IP. It's highly recommended to secure the instance within a private subnet. See the [Advanced Configurations](https://skypilot.readthedocs.io/en/latest/reference/config.html#config-yaml) section of the SkyPilot docs for options to run within VPC and setup private IPs.
+
+## Prompt LoRAX w/ base model
+
+In a separate window, obtain the IP address of the newly created instance:
+
+```shell
+sky status --ip lorax-cluster
+```
+
+Now we can prompt the base model deployment using a simple REST API:
+
+```shell
+IP=$(sky status --ip lorax-cluster)
+
+curl http://$IP:8080/generate \
+    -X POST \
+    -d '{
+        "inputs": "[INST] Natalia sold clips to 48 of her friends in April, and then she sold half as many clips in May. How many clips did Natalia sell altogether in April and May? [/INST]",
+        "parameters": {
+            "max_new_tokens": 64
+        }
+    }' \
+    -H 'Content-Type: application/json'
+```
+
+## Prompt LoRAX w/ adapter
+
+To improve the quality of the response, we can add a single parameter `adapter_id` pointing to a valid LoRA adapter from the [HuggingFace Hub](https://huggingface.co/models).
+
+In this example, we'll use the adapter `vineetsharma/qlora-adapter-Mistral-7B-Instruct-v0.1-gsm8k` that fine-tuned the base model to improve its math reasoning:
+
+```shell
+curl http://$IP:8080/generate \
+    -X POST \
+    -d '{
+        "inputs": "[INST] Natalia sold clips to 48 of her friends in April, and then she sold half as many clips in May. How many clips did Natalia sell altogether in April and May? [/INST]",
+        "parameters": {
+            "max_new_tokens": 64,
+            "adapter_id": "vineetsharma/qlora-adapter-Mistral-7B-Instruct-v0.1-gsm8k"
+        }
+    }' \
+    -H 'Content-Type: application/json'
+```
+
+Here are some other interesting LoRAs to explore with LoRAX:
+
+- [alignment-handbook/zephyr-7b-dpo-lora](https://huggingface.co/alignment-handbook/zephyr-7b-dpo-lora): Mistral-7b fine-tuned on Zephyr-7B dataset with DPO.
+- [timdettmers/guanaco-33b](https://huggingface.co/timdettmers/guanaco-33b): fine-tuned Llama-33b from the QLoRA paper.
+- [tloen/alpaca-lora-7b](https://huggingface.co/tloen/alpaca-lora-7b): Instruction-tuned Llama-7b model using the Alpaca dataset.
```

Comment:
Maybe we should get rid of these and `medalpaca/medalpaca-lora-30b-8bit` below, as they do not work with the Mistral model we launched before.
Or, maybe we group them together and have a note that they should only work with a llama-based model.

Reply:
Fixed.